Feat: description field for env variables

Minamiyama opened the pull request:
## Summary

Add a `description` field to environment variables

## Screenshots

| Before | After |
|--------|-------|
| ![ccdacc241beaaad69bebc20fe841f86](https://github.com/user-attachments/assets/61dc931e-3960-4602-bd70-141d2470d48e) | ![c09f0658c7a1abb6d9a0a8aa5714433](https://github.com/user-attachments/assets/d8a1c15b-27ac-4a2f-8bce-4483128c23eb) |

### Update

new design⬇️⬇️⬇️

https://github.com/user-attachments/assets/1156b003-fdb8-45c4-ac66-18ec6f7b29ac



fixes #21554

## Checklist

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

Minamiyama committed:
feat(workflow_fields): add description field to environment variables

Include description field when formatting SecretVariable and Variable instances in EnvironmentVariableField to provide more context about variables

Minamiyama committed:
i18n

Minamiyama committed:
docs(workflow): add description field to EnvironmentVariable type

Minamiyama committed:
feat(env-panel): add description field to environment variables

Add description field to environment variable modal and display it in the env item component. This allows users to provide additional context for each environment variable.

Minamiyama committed:
style(env-panel): add margin bottom to variable modal value section

Minamiyama committed:
CRLF→LF

Minamiyama committed:
CRLF→LF

Minamiyama committed:
CRLF→LF

Minamiyama committed:
CRLF→LF

crazywoola commented on web/app/components/workflow/nodes/_base/components/variable/utils.ts:
```diff
@@ -462,6 +462,7 @@ const formatItem = (
         return {
           variable: `env.${env.name}`,
           type: env.value_type,
+          des: env.description,
```
```suggestion
          description: env.description,
```

crazywoola commented on web/app/components/workflow/panel/env-panel/variable-modal.tsx:
```diff
@@ -69,6 +71,7 @@ const VariableModal = ({
       setType(env.value_type)
       setName(env.name)
       setValue(env.value_type === 'secret' ? envSecrets[env.id] : env.value)
+      setDes(env.description)
```
```suggestion
      setDescription(env.description)
```

crazywoola reviewed (changes requested):
See comments.

Minamiyama committed:
rename: des→description

Minamiyama committed:
Merge branch 'langgenius:main' into feat/description-field-for-env-var

Minamiyama committed:
Merge branch 'main' into feat/description-field-for-env-var

Minamiyama committed:
Merge branch 'main' into feat/description-field-for-env-var

# Conflicts:
#	web/i18n/ko-KR/workflow.ts

crazywoola commented:
Still needs some time to check with our UI/UX designers.

Minamiyama commented:
> Still needs some time to check with our UI/UX designers.

👌

Minamiyama committed:
refactor(env-panel): improve environment variable item layout and styling

- Move padding to inner container for better hover effects
- Add description section with divider when present
- Use group hover for consistent styling

Minamiyama committed:
style(env-panel): fix line endings and formatting in env-item component
